src/discussions/posts/post/post-shape.js

```diff
@@ -0,0 +1,10 @@
+import PropTypes from 'prop-types';
+
+const postShape = PropTypes.shape({
+  posted_on: PropTypes.string,
+  abuse_flagged: PropTypes.bool,
+  rendered_body: PropTypes.string,
+  author: PropTypes.string,
+});
+
+export default postShape;
```

Comment:
Put the shape in the same file as the component, it can be imported from there. Don't make a separate file.